client/snap_op.go

```diff
@@ -226,15 +226,43 @@ func (client *Client) InstallPath(path, name string, options *SnapOptions) (chan
 		SnapOptions: options,
 	}
 
+	return client.sendLocalSnaps([]string{path}, []*os.File{f}, action)
+}
+
+// InstallPathMany sideloads the snaps with the given paths,
+// returning the UUID of the background operation upon success.
+func (client *Client) InstallPathMany(paths []string, options *SnapOptions) (changeID string, err error) {
+	action := actionData{
+		Action:      "install",
+		SnapOptions: options,
+	}
+
+	var files []*os.File
+	for _, path := range paths {
+		f, err := os.Open(path)
+		if err != nil {
+			for _, openFile := range files {
+				openFile.Close()
+			}
```

Comment:
I think this could go away if deferred closing was moved entirely to this function, see my comment to `sendSnapFiles`.